rplugin/python3/deoplete/sources/look.py

```diff
@@ -14,17 +15,37 @@ def __init__(self, vim):
         self.name = 'look'
         self.mark = '[look]'
         self.min_pattern_length = 4
-        self.executable_look = self.vim.call('executable', 'look')
+
+        def get_look_var(shortname, default):
+            return vim.vars.get('deoplete#look#{}'.format(shortname), default)
+
         self.is_volatile = True
 
+        self.executable_look = self.vim.call('executable', 'look')
+        self.words = get_look_var('words', None)
+        if self.words:
+            self.words = expandvars(expanduser(self.words))
+
+    def _query_look(self, querystring):
+        command = ['look', querystring]
+
+        if self.words is not None:
+            command.append(self.words)
+
+        return subprocess.check_output(command).splitlines()
+
     def gather_candidates(self, context):
         if not self.executable_look:
             return []
 
         try:
-            words = [x.decode(context['encoding']) for x in
-                     subprocess.check_output(
-                         ['look', context['complete_str'][:2]]).splitlines()]
+            # We're adding :2 here to support head prefixed fuzzy search
```

Comment:
hmm, not sure I understand this. So we return all words in the dictionary with the same initial 2 letters? How exactly does deoplete fuzzy search within this list, if no further computation is done on `words` (to my knowledge)? How does this interact with `min_pattern_length`?